.ci/travis/travis.sh

```diff
@@ -555,6 +555,7 @@ verify-no-exception-configs)
 git-status)
   if [ $(git status | grep "Changes not staged for commit" | wc -l) -gt 0 ]; then
     echo "There are changes in files after clone, recheck .gitattributes file"
+    git status
```

Comment:
Sorry looks I was not attentive on merge on this status check.

https://travis-ci.org/github/checkstyle/checkstyle/jobs/685191959

It should not be separate build item, it should be part of post build actions.
Please refactor travis.yml .
Please make second commit, that cause some file creation in git in some existing build to make sure build catch it and fail.

Reply:
Travis builds run in matrix, right?  I was wondering about this.

Reply:
Done.

Reply:
@nmancus1 , please create a file in CI in repository folder during any build item, Travis should fail.
Verification of leftover files in `.ci-temp` is less critical, but please show failure of Travis on it too.

if travis is not failing - verification does not work.

Reply:
> please create a file in CI in repository folder during any build item, Travis should fail
Commit https://github.com/checkstyle/checkstyle/commit/7717c329bd1a7e94bcbdb38ea25eccb667306525 is leaving pmd/ in the root directory of our repo.  This fails git status locally for me.  However, the message that I get when running git status is
```
 On branch issue-8228-validate-git-status
Your branch is up to date with 'origin/issue-8228-validate-git-status'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        pmd/

nothing added to commit but untracked files present (use "git add" to track)

```

So, I should change `git status | grep "Changes not staged for commit"` to something like `git status | grep "Untracked files"? I used the original `grep` command you had written because I assumed that reflected the message that we would get in CI.

Reply:
Done, travis is red

Reply:
Ok, maybe I'm misunderstanding the intention of this validation.  Are we trying to catch untracked files or tracked files that haven't been committed?

Reply:
We need to detect any changes in git repository.